src/lib/floating/Floating.types.ts

```diff
@@ -0,0 +1,39 @@
+import type { ElProps } from '../el'
+
+type FloatingOffset =
+	| number
+	| string
+	| [number, number]
+	| [number, string]
+	| [string, number]
+	| [string, string]
+
+type FloatingPlacement =
+	| 'bottom'
+	| 'center'
+	| 'center-bottom'
+	| 'center-center'
+	| 'center-top'
+	| 'end'
+	| 'end-bottom'
+	| 'end-center'
+	| 'end-top'
+	| 'left'
+	| 'left-bottom'
+	| 'left-center'
+	| 'left-top'
+	| 'right'
+	| 'right-bottom'
+	| 'right-center'
+	| 'right-top'
+	| 'start'
+	| 'start-bottom'
+	| 'start-center'
+	| 'start-top'
+	| 'top'
+
+export interface FloatingProps extends Partial<Omit<ElProps, 'fixed' | 'offset'>> {
```

Comment:
@abdolian what do you think about `floatingOffset` prop name instead of `offset`?

and for `fixed` prop, I think there is no need to define `fixed` in this component. El has `position="fixed"` prop which we can use it.

Reply:
1- We define offset on the floating component and floatingOffset name has extra word. We can use synonym instead of floatingOffset.
2- That's correct, we can remove it.
@TheHadiAhmadi